drivers/sensor/ds18b20/ds18b20.c

```diff
@@ -0,0 +1,268 @@
+static const uint16_t measure_wait_ms[4] = {94, 188, 376, 750};
+
+static inline void ds18b20_temperature_from_raw(uint8_t *temp_raw,
+						struct sensor_value *val)
+{
+	uint16_t temp = sys_get_le16(temp_raw);
+
+	val->val1 = (temp & 0x0FFF) >> 4;
+	val->val2 = (temp & 0x000F) * 10000;
+}
+
+/*
+ * Write scratch pad, read back, then copy to eeprom
+ */
+static int ds18b20_write_scratchpad(const struct device *dev,
+				    struct ds18b20_scratchpad scratchpad)
+{
+
+	struct ds18b20_data *data = dev->data;
+	const struct device *bus = ds18b20_bus(dev);
+	uint8_t sp_data[4] = {
+		DS18B20_CMD_WRITE_SCRATCHPAD,
+		scratchpad.alarm_temp_high,
+		scratchpad.alarm_temp_low,
+		scratchpad.config
+	};
+
+	return w1_write_read(bus, &data->config,  sp_data, sizeof(sp_data),
+			     NULL, 0);
+}
+
+static int ds18b20_read_scratchpad(const struct device *dev,
+				   struct ds18b20_scratchpad *scratchpad)
+{
+	struct ds18b20_data *data = dev->data;
+	const struct device *bus = ds18b20_bus(dev);
+	uint8_t cmd =  DS18B20_CMD_READ_SCRATCHPAD;
+
+	return w1_write_read(bus, &data->config,  &cmd, 1,
+			     (uint8_t *)&scratchpad[0], 9);
+}
+
+/* Starts sensor temperature conversion without waiting for completion. */
+static int ds18b20_temperature_convert(const struct device *dev)
+{
+	int ret;
+	struct ds18b20_data *data = dev->data;
+	const struct device *bus = ds18b20_bus(dev);
+
+	ret = w1_reset_select(bus, &data->config);
+	if (ret != 0) {
+		return ret;
+	}
+
+	return  w1_write_byte(bus, DS18B20_CMD_CONVERT_T);
+}
+
+/* Read power supply status from the sensor. */
+static int ds18b20_read_power_supply(const struct device *dev)
+{
+	struct ds18b20_data *data = dev->data;
+	const struct device *bus = ds18b20_bus(dev);
+	int ret;
+	uint8_t cmd = DS18B20_CMD_READ_POWER_SUPPLY;
+
+	ret = w1_write_read(bus, &data->config,  &cmd, 1, NULL, 0);
+	if (ret != 0) {
+		return ret;
+	}
+
+	ret = w1_read_bit(bus);
+	if (ret < 0) {
+		return ret;
+	}
+	data->power_supply = (bool)ret;
+	return 0;
+}
+
+/*
+ * Write resolution into configuration struct,
+ * but don't write it to the sensor yet.
+ */
+static int ds18b20_set_resolution(const struct device *dev, uint8_t resolution)
+{
+	struct ds18b20_data *data = dev->data;
+
+	data->scratchpad.config &= ~DS18B20_RESOLUTION_MASK;
+	data->scratchpad.config |= DS18B20_RESOLUTION(resolution);
+
+	return 0;
+}
+
+static int ds18b20_sample_fetch(const struct device *dev,
+				enum sensor_channel chan)
+{
+	const struct ds18b20_config *cfg = dev->config;
+	struct ds18b20_data *data = dev->data;
+	int status;
+	int ret;
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL ||
+			chan == SENSOR_CHAN_AMBIENT_TEMP);
+
+	if (!data->lazy_loaded) {
+		status = ds18b20_configure(dev);
+		if (status < 0) {
+			return status;
+		}
+		data->lazy_loaded = true;
+	}
+
+	ret = ds18b20_temperature_convert(dev);
+	if (ret < 0) {
+		LOG_DBG("W1 fetch error");
+		return -EIO;
+	}
```

Comment:
Since `ret` is not returned anywhere it does not seem to be needed; there is also `status` variable that shares the same purpose.

Reply:
I changed it to assign the value to status, and return it afterwards.